Model/Throttle.php

```diff
@@ -123,8 +123,13 @@ public function check($key, $allowed = 1) {
 		if (empty($allowed)) {
 			$allowed = 1;
 		}
-		$this->purge();
-		$count = $this->find('count', array( 'conditions' => compact('key') ));
+		register_shutdown_function(function() {
+			$this->purge();
```

Comment:
Starting w/ 5.4 you can use $this in anonymous functions: http://php.net/manual/en/migration54.new-features.php

I've tested this through a controller calling ->throttle and it worked properly